examples_tests/20.Megatexture/main.cpp

```diff
@@ -195,20 +182,37 @@ vec3 unpackVirtualUV(in uvec2 texData)
 }
 vec2 unpackSize(in uvec2 texData)
 {
-	return vec2(texData.x&0xffffu,texData.x>>16u);//363
+	return vec2(texData.x&0xffffu,texData.x>>16u);
 }
-vec4 textureVT(in uvec2 _texData, in vec2 uv, mat2 dUV)
+#define WRAP_REPEAT 0u
+#define WRAP_CLAMP 1u
+#define WRAP_MIRROR 2u
+float wrapTexCoord(float tc, in uint mode)
+{
+    switch (mode)
+    {
+    case WRAP_REPEAT: tc = fract(tc); break;
+    case WRAP_CLAMP:  tc = clamp(tc, 0.0, 1.0); break;
+    case WRAP_MIRROR: tc = 1.0 - abs(mod(tc,2.0)-1.0); break;
+    default: break;
+    }
```

Comment:
I've had a really bad time with performance and switches in my GPU raytracingwork... use if statements, also I gave you a GLSL snippet on discord on how to implement the wrapping modes with the least divergence (reusing a few ops)

Reply:
TODO